servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/PublisherFlatMapMerge.java

```diff
@@ -0,0 +1,694 @@
+                            min(parent.source.maxMappedDemand - prevOutstandingDemand, n));
+                    if (quotaToUse == 0) {
+                        if (outstandingDemandUpdater.compareAndSet(this, prevOutstandingDemand,
+                                prevOutstandingDemand)) {
+                            return 0;
+                        }
+                    } else if (outstandingDemandUpdater.compareAndSet(this, prevOutstandingDemand,
+                            prevOutstandingDemand + quotaToUse)) {
+                        pendingDemandUpdater.addAndGet(this, quotaToUse);
+                        for (;;) {
+                            final Subscription prevSubscription = subscription;
+                            if (prevSubscription == null) {
+                                if (subscriptionUpdater.compareAndSet(this, null, REQUEST_PENDING)) {
+                                    break;
+                                }
+                            } else if (prevSubscription == CANCELLED || prevSubscription == CANCEL_PENDING) {
+                                break;
+                            } else if (prevSubscription == REQUEST_PENDING || prevSubscription == PROCESSING_ONNEXT ||
+                                    prevSubscription == PROCESSING_REQUEST) {
+                                if (subscriptionUpdater.compareAndSet(this, prevSubscription, REQUEST_PENDING)) {
+                                    break;
+                                }
+                            } else if (subscriptionUpdater.compareAndSet(this, prevSubscription, PROCESSING_REQUEST)) {
+                                doRequestMore(prevSubscription);
+                                break;
+                            }
+                        }
+                        return quotaToUse;
+                    }
+                }
+            }
+
+            private void doRequestMore(Subscription prevSubscription) {
+                long availableRequestN = pendingDemandUpdater.getAndSet(this, 0);
+                for (;;) {
+                    if (availableRequestN > 0) {
+                        try {
+                            prevSubscription.request(availableRequestN);
+                        } catch (Throwable cause) {
+                            subscription = CANCELLED;
+                            cancelAndGiveBack(prevSubscription);
+                            throw cause;
+                        }
+                    }
+
+                    final Subscription afterLockSubscription = subscription;
+                    if (afterLockSubscription == PROCESSING_ONNEXT || afterLockSubscription == CANCELLED) {
+                        // onNext is allowed to interrupt, and will handle any pending cancel/request events
+                        // after it completes (otherwise we may drop signals).
+                        break;
+                    } else if (afterLockSubscription == CANCEL_PENDING) {
```

Comment:
nit: you could remove the `else`

Reply:
I will keep the `else` as it clarifies expectations, avoids accidental fall-through issues as code evolves, and keeps code more vertically compact.